gremlin-go/driver/graph.go

```diff
@@ -120,3 +121,63 @@ func (p *Path) GetPathObject(key string) (interface{}, error) {
 		return nil, fmt.Errorf("path does not contain a label of '%s'", key)
 	}
 }
+
+// SetTemplate describes the necessary methods that need to be implemented to make a user-defined Set implementation.
+// Set is the default implementation that should be used but SetTemplate can be used for a custom implementation if needed.
+type SetTemplate interface {
+	convertSetToSlice() ([]interface{}, error)
+	convertSliceToSet([]interface{}) (SetTemplate, error)
+}
+
+// Set is a default custom declaration since Go does not natively provide this feature.
+type Set struct {
+	Objects []interface{}
+}
+
+func (s *Set) convertSetToSlice() ([]interface{}, error) {
+	return s.Objects, nil
+}
+
+func (s *Set) convertSliceToSet(in []interface{}) (SetTemplate, error) {
+	slice, err := interfaceToSlice(in)
+	if err != nil {
+		return nil, err
+	}
+	s.Objects = slice
+	return s, nil
+}
+
+// Construct a new Set with filtered values
+func newSet(in interface{}) *Set {
```

Comment:
We only need one method for instantiation which is exported.

`NewSet(args... interface{}) *GremlinSet`

This will iterate through the args and only add things to the underlying objects slice if they are not currently in the slice.

Reply:
Agree but NewGremlinSet instead? In case we decide to add some other implementation later.